src/Bicep.Core/Syntax/ProviderDeclarationSyntax.cs

```diff
@@ -41,14 +42,14 @@ public ProviderDeclarationSyntax(IEnumerable<SyntaxBase> leadingNodes, Token key
 
         public IdentifierSyntax? Alias => (this.AsClause as AliasAsClauseSyntax)?.Alias;
 
-        private StringSyntax ProviderPath => SyntaxFactory.CreateStringLiteral($@"{OciArtifactReferenceFacts.Scheme}:{LanguageConstants.BicepPublicMcrRegistry}/bicep/providers/{this.Specification.Name}:{this.Specification.Version}");
-
         public override TextSpan Span => TextSpan.Between(this.Keyword, TextSpan.LastNonNull(this.SpecificationString, this.WithClause, this.AsClause));
 
         SyntaxBase IForeignArtifactReference.ReferenceSourceSyntax => ProviderPath;
 
         public override void Accept(ISyntaxVisitor visitor) => visitor.VisitProviderDeclarationSyntax(this);
 
+        private StringSyntax ProviderPath => SyntaxFactory.CreateStringLiteral($@"{OciArtifactReferenceFacts.Scheme}:{FeatureProvider.ReadEnvVar("BICEP_REGISTRY_FQDN", LanguageConstants.BicepPublicMcrRegistry)}/bicep/providers/{this.Specification.Name}:{this.Specification.Version}");
```

Comment:
Is this **just** for testing? Or is this intended to serve a bunch of other scenarios?

The reason why I'm asking is that in the fullness of time you'll probably want a list of search paths that might contain multiple repos.

For example there's only one registry that can be provided here. If I'm developing my own provider, then do I need to re-host all of the other providers I use in my registry?

Reply:
This is a temporary thing, in future PRs we will make the registry configurable with aliases in a similar way modules work